akka-http/server/src/main/scala/endpoints/akkahttp/server/Urls.scala

```diff
@@ -60,6 +69,8 @@ trait Urls extends algebra.Urls {
   // Paths
   // ********
 
+  override implicit def uuidSegment: Segment[UUID] = JavaUUID
```

Comment:
Are we sure that this behaves like the default (inherited) implementation?

Reply:
In particular, the current server implementation assumes that a segment with the wrong format (e.g. a string which is not a valid UUID) does not match the path description.

Reply:
I think the two are equivalent.

In `algebra.Urls`, we have:

```
  implicit def uuidSegment: Segment[UUID] =
    refineSegment[String, UUID](stringSegment)((s: String) => Try.apply(UUID.fromString(s)).toOption)(_.toString)
```

which in the context of `server.Urls`, we have:

```
  def refineSegment[A, B](sa: Segment[A])(f: A => Option[B])(g: B => A): Segment[B] =
    sa.tflatMap[Tuple1[B]]((a: Tuple1[A]) => f(a._1).map(Tuple1.apply))
```

So this becomes:

```
  implicit def uuidSegment: Segment[UUID] =
stringSegment.tflatMap[Tuple1[UUID]](a: Tuple1[String]) => Try.apply(UUID.fromString(a._1)).toOption)(_.toString)).map(Tuple1.apply))
```

Compare this with:

```
  val JavaUUID: PathMatcher1[UUID] =
    PathMatcher("""[\da-fA-F]{8}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{12}""".r) flatMap { string ⇒
      try Some(UUID.fromString(string))
      catch { case _: IllegalArgumentException ⇒ None }
    }
```

If anything, the `JavaUUUID` version does redundant parsing of the UUID string once via the regex `PathMatcher` and if that succeeds, then with `UUID.fromString(_)`. In the refine case, the UUID string is parsed only once in `UUID.fromString(_)`. 

I'll delete the redundant override.